src/App/VPSScan/Scan/Core.hs

```diff
@@ -138,32 +114,23 @@ getProjectScanFilters FossaOpts{..} locator = runHTTP $ do
   pure (responseBody resp)
 
 overrideScanFilters :: (Has (Lift IO) sig m, Has Diagnostics sig m, Has Trace sig m) => VPSOpts -> Locator -> m (VPSOpts, Bool)
-overrideScanFilters vpsOpts@VPSOpts{..} locator
-  | (unFilterExpressions filterBlob) == "[]" = do
-    trace "[All] Fetching scan file filter from FOSSA"
-    overrideFilters <- getProjectScanFilters fossa locator
-    let overrideFilterBlob = toFilterExpressions overrideFilters
-    trace $ unpack $ "[All] Using filter: " <> (unFilterExpressions overrideFilterBlob)
-    pure (VPSOpts fossa projectName userProvidedRevision skipIprScan overrideFilterBlob, True)
-  | otherwise = do
-    trace "[All] Scan file filters provided locally"
-    pure (vpsOpts, False)
-
+overrideScanFilters vpsOpts@VPSOpts { fileFilter = (FilterExpressions []) } locator = do
+  let VPSOpts{..} = vpsOpts
+  trace "[All] Fetching scan file filter from FOSSA"
+  overrideFilters <- getProjectScanFilters fossa locator
+  trace $ unpack $ "[All] Using scan file filter: " <> encodeFilterExpressions overrideFilters
+  pure (VPSOpts fossa projectName userProvidedRevision skipIprScan overrideFilters, True)
```

Comment:
I'm pretty sure it's equivalent, and I think it would be more clear that all we're changing is `fileFilter` if you did:

```haskell
pure (vpsOpts{fileFilter = overrideFilters}, True)
```